Tests/FoundationEssentialsTests/StringTests.swift

```diff
@@ -327,3 +327,1925 @@ final class StringTests : XCTestCase {
+    }
+
+    func test_capitalizedString() {
+        expectLocalizedEquality(
+            "Foo Foo Foo Foo",
+            { loc in "foo Foo fOO FOO".capitalized(with: loc) })
+
+        expectLocalizedEquality("Жжж", { loc in "жжж".capitalized(with: loc) })
+
+        expectEqual(
+            "Foo Foo Foo Foo",
+            "foo Foo fOO FOO".capitalized(with: nil))
+        expectEqual("Жжж", "жжж".capitalized(with: nil))
+
+        //
+        // Special casing.
+        //
+
+        // U+0069 LATIN SMALL LETTER I
+        // to upper case:
+        // U+0049 LATIN CAPITAL LETTER I
+        expectLocalizedEquality(
+            "Iii Iii",
+            { loc in "iii III".capitalized(with: loc) }, "en")
+
+        // U+0069 LATIN SMALL LETTER I
+        // to upper case in Turkish locale:
+        // U+0130 LATIN CAPITAL LETTER I WITH DOT ABOVE
+        expectLocalizedEquality(
+            "İii Iıı",
+            { loc in "iii III".capitalized(with: loc) }, "tr")
+    }
+
+    func test_caseInsensitiveCompare() {
+        expectEqual(ComparisonResult.orderedSame,
+                    "abCD".caseInsensitiveCompare("AbCd"))
+        expectEqual(ComparisonResult.orderedAscending,
+                    "abCD".caseInsensitiveCompare("AbCdE"))
+
+        expectEqual(ComparisonResult.orderedSame,
+                    "абвг".caseInsensitiveCompare("АбВг"))
+        expectEqual(ComparisonResult.orderedAscending,
+                    "абВГ".caseInsensitiveCompare("АбВгД"))
+    }
+
+    func test_commonPrefix() {
+        expectEqual("ab",
+                    "abcd".commonPrefix(with: "abdc", options: []))
+        expectEqual("abC",
+                    "abCd".commonPrefix(with: "abce", options: .caseInsensitive))
+
+        expectEqual("аб",
+                    "абвг".commonPrefix(with: "абгв", options: []))
+        expectEqual("абВ",
+                    "абВг".commonPrefix(with: "абвд", options: .caseInsensitive))
+    }
+
+    func test_compare() {
+        expectEqual(ComparisonResult.orderedSame,
+                    "abc".compare("abc"))
+        expectEqual(ComparisonResult.orderedAscending,
+                    "абв".compare("где"))
+
+        expectEqual(ComparisonResult.orderedSame,
+                    "abc".compare("abC", options: .caseInsensitive))
+        expectEqual(ComparisonResult.orderedSame,
+                    "абв".compare("абВ", options: .caseInsensitive))
+
+        do {
+            let s = "abcd"
+            let r = s.index(after: s.startIndex)..<s.endIndex
+            expectEqual(ComparisonResult.orderedSame,
+                        s.compare("bcd", range: r))
+        }
+        do {
+            let s = "абвг"
+            let r = s.index(after: s.startIndex)..<s.endIndex
+            expectEqual(ComparisonResult.orderedSame,
+                        s.compare("бвг", range: r))
+        }
+
+        expectEqual(ComparisonResult.orderedSame,
+                    "abc".compare("abc", locale: Locale.current))
```

Comment:
Should we make this an explicit locale so that the results of this test don't depend on the test host device's locale (in actually, I guess the current locale would probably never actually affect this comparison since they're equal, but might be a good change for correctness' sake?)

Reply:
Good catch. Definitely